modules/timeoutRtdProvider.js

```diff
@@ -0,0 +1,173 @@
+
+import { submodule } from '../src/hook.js';
+import * as ajax from '../src/ajax.js';
+import * as utils from '../src/utils.js';
+import { getGlobal } from '../src/prebidGlobal.js';
+
+const SUBMODULE_NAME = 'timeout';
+
+// this allows the stubbing of functions during testing
+export const timeoutRtdFunctions = {
+  getDeviceType,
+  getConnectionSpeed,
+  checkVideo,
+  calculateTimeoutModifier,
+  handleTimeoutIncrement
+};
+
+function getDeviceType() {
+  const userAgent = window.navigator.userAgent.toLowerCase();
+  if ((/ipad|android 3.0|xoom|sch-i800|playbook|tablet|kindle/i.test(userAgent))) {
+    return 5; // tablet
+  }
+  if ((/iphone|ipod|android|blackberry|opera|mini|windows\sce|palm|smartphone|iemobile/i.test(userAgent))) {
+    return 4; // mobile
+  }
+  return 2; // personal computer
+}
+
+function checkVideo(adUnits) {
+  return adUnits.some((adUnit) => {
+    return adUnit.mediaTypes && adUnit.mediaTypes.video;
```

Comment:
might be useful to consider outstream multiformat units differently?

Reply:
Yeah, though I think we'll just save that for a future version of this module